R/checks.R

```diff
@@ -57,51 +58,63 @@ check_user_rights <- function(
 #' \code{REDCapR::redcap_read_oneshot()$data}
 #'
 #' @importFrom rlang .data
-#' @importFrom dplyr slice pull %>% select
-#' @importFrom tidyr drop_na
-#' @importFrom tidyselect everything
+#' @importFrom dplyr %>% select mutate case_when
+#' @importFrom purrr map2
+#' @importFrom tidyselect any_of
 #'
 #' @keywords internal
 
 
 check_repeat_and_nonrepeat <- function(db_data) {
 
-  # Loop through dataframe columns then rows to determine if a variable has data
-  # and appears alongside redcap_repeat_instance when it is both NA and not NA
-
-  #start at 5 to skip record id, redcap event, instrument, and instance vars
-  for (i in 5:length(names(db_data))) {
-
-    nonrepeating <- NA
-    repeating <- NA
-
-    for (j in seq_len(nrow(db_data))){
+  # This check function looks for potential repeat/nonrepeat behavior using the
+  # steps below:
+  # 1) Define standard columns that don't need checking and remove those from
+  #    analysis (i.e. "safe columns").
+  # 2) Create a dummy column for each remaining column and use case_when to
+  #    to assign whether the column demonstrates repeating, nonrepeating, or
+  #    indeterminant behavior per row. Indetemrinant would be the case for
+  #    data not yet filled out.
+  # 3) Use a mapping function to determine if any dummy columns contain both
+  #    "repeating" AND "nonrepeating" declarations, if so error out.
+
+  # Step (1)
+  safe_cols <- c(names(db_data)[1], "redcap_event_name",
```

Comment:
Arm?

Reply:
Not sure I follow, the standard vars are `redcap_event_name`, `redcap_repeat_instrument`, `redcap_repeat_instance`:
![image](https://user-images.githubusercontent.com/38384694/196563870-b2962d9d-80f0-4f42-88c2-14faf3ac1247.png)

We manually split `event_name` into `event` and `arm` later on.